src/user/controllers/user.controller.ts

```diff
@@ -1,25 +1,46 @@
+import { CreateUserDto } from 'src/user/dto/create-user.dto';
 import { Controller, Get, Post, Body, Patch, Param, Delete, Query } from '@nestjs/common';
 import { UserService } from '../services/user.service';
-import { CreateUserDto } from '../dto/create-user.dto';
 import { UpdateUserDto } from '../dto/update-user.dto';
+import { DuplicateRecordException } from 'src/filters/duplicate-record-exception.filter';
+import { NotFoundError } from 'rxjs';
+import { NotFoundExceptionFilter } from 'src/filters/not-found-exception.filter';
 
 @Controller('user')
 export class UserController {
   constructor(private readonly userService: UserService) {}
 
   @Post()
-  create(@Body() user: CreateUserDto) {
-    return this.userService.create(user);
+  async create(@Body() user: CreateUserDto) {
+    const isExistUser = await this.userService.findOne(user.email);
+    if(isExistUser) {
+      throw new DuplicateRecordException();
+    }
+    this.userService.create(user);
+    return 'Usuário criado com sucesso!'
   }
 
   @Get()
-  findOne(@Query('email') email: string) {
-    return this.userService.findOne(email);
+  async findOne(@Query('email') email: string) {
+    if (await this.userService.findOne(email)) {
+      return this.userService.findOne(email);
```

Comment:
pensando se não seria melhor salvar em uma variável para não ter que fazer a chamada duas vezes:

```suggestion
    const user = await this.userService.findOne(email)
    if (user) {
      return user;
```